contracts/margin/external/AuctionProxy.sol

```diff
@@ -0,0 +1,176 @@
+/*
+
+    Copyright 2018 dYdX Trading Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+*/
+
+pragma solidity 0.4.24;
+pragma experimental "v0.5.0";
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { Margin } from "../Margin.sol";
+import { ZeroExExchangeInterface } from "../../external/0x/ZeroExExchangeInterface.sol";
+import { MathHelpers } from "../../lib/MathHelpers.sol";
+import { TokenInteract } from "../../lib/TokenInteract.sol";
+import { ZeroExV1Parser } from "../../lib/ZeroExV1Parser.sol";
+
+
+/**
+ * @title AuctionProxy
+ * @author dYdX
+ *
+ * Contract for allowing anyone to close a Dutch Auction
+ */
+contract AuctionProxy is
+    ZeroExV1Parser
+{
+    using TokenInteract for address;
+    using SafeMath for uint256;
+
+    // ============ Structs ============
+
+    struct Position {
+        address heldToken;
+        address owedToken;
+        address tokenContract;
```

Comment:
don't call this `tokenContract`, this could be used by nontokenized positions as well. just `owner` is fine